src/Aspire.Dashboard/Components/Pages/Metrics.razor.cs

```diff
@@ -167,10 +169,8 @@ private Task HandleSelectedApplicationChangedAsync()
             }
         }
 
-        // On mobile, we actually *do* want to update the selected application immediately, since it will affect the tree of possible
-        // metrics to select from. So, we must also immediately close the window since the closing behavior is necessary if the url has changed.
-        var isChangeInToolbar = ViewportInformation.IsDesktop;
-        return this.AfterViewModelChangedAsync(_contentLayout, isChangeInToolbar: isChangeInToolbar);
+        await this.AfterViewModelChangedAsync(_contentLayout, waitToApplyMobileChange: true);
+        _treeMetricSelector?.OnResourceChanged();
```

Comment:
Do we need to call StateHasChanged on component or could StateHasChanged be called here?

Reply:
Unfortunately, calling  `StateHasChanged` on Metrics doesn't work, since this is in a render fragment inside a different component. That was the motivation initially behind the ref